plugins/optimization-detective/optimization.php

```diff
@@ -80,14 +80,38 @@ static function ( string $output, ?int $phase ): string {
  * @access private
  */
 function od_maybe_add_template_output_buffer_filter(): void {
-	if (
-		! od_can_optimize_response() ||
-		od_is_rest_api_unavailable() ||
-		isset( $_GET['optimization_detective_disabled'] ) // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-
-	) {
+	$conditions = array(
+		array(
+			'test'   => od_can_optimize_response(),
+			'reason' => __( 'Page is not optimized because od_can_optimize_response() returned false. This can be overridden with the od_can_optimize_response filter.', 'optimization-detective' ),
+		),
+		array(
+			'test'   => ! ( od_is_rest_api_unavailable() && ( wp_get_environment_type() !== 'local' || class_exists( 'Test_OD_Optimization' ) ) ),
```

Comment:
I'd prefer that we "resolve" this negation, as it's more complex to parse in its current form.

I _think_ I got this one right? 🤔 😂 
```suggestion
			'test'   => ! od_is_rest_api_unavailable() || wp_get_environment_type() === 'local' && ! class_exists( 'Test_OD_Optimization' ),
```

Alternatively we keep the parentheses around the second condition for clarity:
```suggestion
			'test'   => ! od_is_rest_api_unavailable() || ( wp_get_environment_type() === 'local' && ! class_exists( 'Test_OD_Optimization' ) ),
```

Reply:
I asked PhpStorm to refactor this and it produced the same result as you. 👍🏻